999-storage-import-zipped-vhds/azuredeploy.json

```diff
@@ -0,0 +1,173 @@
untBlobContainerSasToken = $Destination \n  \n $ResourceGroupName = ${Env:ResourceGroupName} \n  \n ##### Variables \n  \n $base_path='/mnt/azscripts' \n $temp_path='{0}/temp' -f $base_path \n New-Item -Type Directory -ErrorAction SilentlyContinue $temp_path \n  \n $timestamp = Get-Date -Format 'yyMMddHHmmss' \n  \n ##### Fetch prerequisites \n  \n Invoke-WebRequest -Uri $azcopyArchiveUrl -Outfile azcopy-linux.tar.gz \n tar xf azcopy-linux.tar.gz \n Move-Item azcopy_linux_*/azcopy . -Force \n  \n ##### Check destination is actually writable or fail \n  \n $testFileName = 'Test-{0}' -f $timestamp \n New-Item -Force -Type File -Name $testFileName \n ./azcopy copy $testFileName \"$uriWritableStorageAccountBlobContainerSasToken\" \n if (! $?) { \n   Throw 'Cannot upload into the Storage Account Container. Please check the Destination parameter contains the URI of Storage Account Container with the SAS Token' \n } \n  \n ##### Download for processing \n  \n Write-Output 'Increase file share quota to 4TB instead of 2GB before downloading and expanding' \n $QuotaGiB = '4096' \n Get-AzStorageAccount -ResourceGroupName $ResourceGroupName | Where-Object { $_.StorageAccountName -like '*azscripts' } | ForEach-Object { \n   $StorageAccount = $_ \n   Get-AzStorageShare -Context $StorageAccount.Context | ForEach-Object {  \n     $Share = $_ \n     Set-AzStorageShareQuota -ShareName $Share.Name -Context $StorageAccount.Context -Quota $QuotaGiB \n   } \n } \n  \n  \n Write-Output 'Download the ZIP archive' \n $archive_path='{0}/archive-{1}.zip' -f $temp_path,$timestamp \n # Accelerate file download (initially using Invoke-WebRequest) \n (New-Object System.Net.WebClient).DownloadFile($urlVHDZipArchive, $archive_path) \n  \n Write-Output 'Expand locally' \n $expanded_archive_path ='{0}/expanded-{1}' -f $temp_path,$timestamp \n New-Item -Type Directory -ErrorAction SilentlyContinue $expanded_archive_path \n Expand-Archive -Path $archive_path -DestinationPath $expanded_archive_path -Force \n  \n ##### Upload to Azure Storage account \n  \n Write-Output 'Upload extracted VHD file(s)' \n $vhd_filepath='{0}/*.vhd' -f  $expanded_archive_path \n # Temp to fix copy issue due to low resources ./azcopy copy \"$vhd_filepath\" \"$uriWritableStorageAccountBlobContainerSasToken\" --put-md5 \n ./azcopy copy \"$vhd_filepath\" \"$uriWritableStorageAccountBlobContainerSasToken\" \n  \n ##### Output \n  \n $DeploymentScriptOutputs = @{} \n $DeploymentScriptOutputs['vhdBlobUriList'] = @{} \n $DeploymentScriptOutputs['vhdBlobSHA256List'] = @{} \n  \n $StorageAccountContainerUri = ($uriWritableStorageAccountBlobContainerSasToken -split [Regex]::Escape('?'))[0] \n Get-Item $vhd_filepath | % { \n   $VHDBlobName = $_.Name \n   $vhdBlobUri = \"$StorageAccountContainerUri/$VHDBlobName\" \n   $hash = (Get-FileHash -Algorithm SHA256 -Path $_).Hash \n   $DeploymentScriptOutputs['vhdBlobUriList'][$_.Name] = $vhdBlobUri \n   $DeploymentScriptOutputs['vhdBlobSHA256List'][$_.Name] = $hash \n } \n "
```

Comment:
extract this to a file since you already have script files in the sample

Reply:
This Inlining+Onelining trick is to workaround following issues. I kept the .ps1 in scripts folder for readability and preprocessing:
- with a relative reference to the script file (using artifactlocation)  then local template deployment is no more possible and the pipeline would also raise error
- multi-lines scriptContent creates parsing issue when using template in Azure Portal

What is you recommendation?

Reply:
Following the advice about the same issue with PR 10646, i removed the file and kept the scriptContent